src/shims/unix/linux/epoll.rs

```diff
@@ -11,24 +13,56 @@ use self::shims::unix::fd::FileDescriptor;
 #[derive(Clone, Debug, Default)]
 struct Epoll {
     /// The file descriptors we are watching, and what we are watching for.
-    file_descriptors: FxHashMap<i32, EpollEvent>,
+    interest_list: BTreeMap<(WeakFileDescriptor, i32), Rc<EpollEvent>>,
```

Comment:
The comment is incomplete -- there's an odd `i32` also sitting here whose meaning is not explained.